src/ConfigParser.cpp

```diff
@@ -48,63 +176,55 @@ void ConfigParser::parseConfig(const std::string& filename)
 					else
 						currentSection.clear();
 				}
```

Comment:
Just a general note, there is quite a lot of nesting here! we could just check if the sectionstack is empty and throw in that case, where we would not have to have so many brackets etc! no need for a change, small thing and more for longer lasting projects but just noting!